balancer/randomsubsetting/randomsubsetting_test.go

```diff
@@ -0,0 +1,297 @@
+			gotCfg, gotErr := parser.ParseConfig(json.RawMessage(test.input))
+			// Substring match makes this very tightly coupled to the
+			// internalserviceconfig.BalancerConfig error strings. However, it
+			// is important to distinguish the different types of error messages
+			// possible as the parser has a few defined buckets of ways it can
+			// error out.
+			if (gotErr != nil) != (test.wantErr != "") {
+				t.Fatalf("ParseConfig(%v) = %v, wantErr %v", test.input, gotErr, test.wantErr)
+			}
+			if gotErr != nil && !strings.Contains(gotErr.Error(), test.wantErr) {
+				t.Fatalf("ParseConfig(%v) = %v, wantErr %v", test.input, gotErr, test.wantErr)
+			}
+			if test.wantErr != "" {
+				return
+			}
+			if diff := cmp.Diff(gotCfg, test.wantCfg); diff != "" {
+				t.Fatalf("ParseConfig(%v) got unexpected output, diff (-got +want): %v", test.input, diff)
+			}
+		})
+	}
+}
+
+func setupBackends(t *testing.T, backendsCount int) ([]resolver.Address, func()) {
+	t.Helper()
+
+	backends := make([]*stubserver.StubServer, backendsCount)
+	addresses := make([]resolver.Address, backendsCount)
+	for i := 0; i < backendsCount; i++ {
+		backend := &stubserver.StubServer{
+			EmptyCallF: func(ctx context.Context, in *testpb.Empty) (*testpb.Empty, error) {
+				return &testpb.Empty{}, nil
+			},
+		}
+		if err := backend.StartServer(); err != nil {
+			t.Fatalf("Failed to start backend: %v", err)
+		}
+		t.Logf("Started good TestService backend at: %q", backend.Address)
+		backends[i] = backend
+		addresses[i] = resolver.Address{
+			Addr: backend.Address,
+		}
+	}
+
+	cancel := func() {
+		for _, backend := range backends {
+			backend.Stop()
+		}
+	}
+	return addresses, cancel
+}
+
+func setupClients(t *testing.T, clientsCount int, subsetSize int, addresses []resolver.Address) ([]testgrpc.TestServiceClient, func()) {
+	t.Helper()
+
+	clients := make([]testgrpc.TestServiceClient, clientsCount)
+	ccs := make([]*grpc.ClientConn, clientsCount)
+	var err error
+
+	for i := 0; i < clientsCount; i++ {
+		mr := manual.NewBuilderWithScheme("subsetting-e2e")
+		jsonConfig := fmt.Sprintf(`
+		{
+		  "loadBalancingConfig": [
+			{
+			  "random_subsetting": {
+				"subset_size": %d,
+				"child_policy": [{"round_robin": {}}]
+			  }
+			}
+		  ]
+		}`, subsetSize)
+
+		sc := internal.ParseServiceConfig.(func(string) *serviceconfig.ParseResult)(string(jsonConfig))
+		mr.InitialState(resolver.State{
+			Addresses:     addresses,
+			ServiceConfig: sc,
+		})
+
+		ccs[i], err = grpc.Dial(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithTransportCredentials(insecure.NewCredentials()))
+		if err != nil {
+			t.Fatalf("grpc.Dial() failed: %v", err)
+		}
+		clients[i] = testgrpc.NewTestServiceClient(ccs[i])
+	}
+
+	cancel := func() {
+		for _, cc := range ccs {
+			cc.Close()
+		}
+	}
+	return clients, cancel
+}
+
+func checkRoundRobinRPCs(ctx context.Context, t *testing.T, clients []testgrpc.TestServiceClient, subsetSize int, maxDiff int) {
```

Comment:
We have something like this already: https://github.com/grpc/grpc-go/blob/e8167363406cc5a2da7befab1bfc5e4c21bdd264/internal/testutils/roundrobin/roundrobin.go#L85

Would you be able to use that instead? Thanks